src/util/bip32.rs

```diff
@@ -46,11 +46,73 @@ impl Default for Fingerprint {
     fn default() -> Fingerprint { Fingerprint([0; 4]) }
 }
 
+/// Script type for addresses
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum ScriptType {
```

Comment:
I think `ScriptType` is a very confusing name. First, it's scriptPubkey type, second, it may mix with other cases where script type name may be used in the future. So I propose to rename it to something like `UsageScenario`